AEP/aep-0000y.md

```diff
@@ -0,0 +1,74 @@
+# AEP : ArmoniK's Domain Definition
+
+|                   |ArmoniK Enhancement Proposal|
+---:                |:---
+**AEP**             | 
+**Title**           | Architecture for ArmoniK Extesion development
+**Author**          | Italo Aguiar <<iaguiar@aneo.fr>>, Quentin Delamea <<qdelamea@aneo.fr>>
+**Status**          | Draft
+**Type**            | Standard
+**Creation Date**   | 2024-06-04
+
+# Abstract
+
+This AEP proposes a comprehensive definition of ArmoniK's domain. A domain is an area of interest or control within our system. Our goal is to clearly define the core domain entities to facilitate further development.
+
+# Motivation
+
+The purpose of this document is to establish a clear and consistent definition of ArmoniK's domain, enabling other documents and standards to effectively reuse these concepts.
+
+# Rationale
+
+To describe more complex features of ArmoniK accurately, it is crucial to have a shared vocabulary and avoid redundancy. While some definitions already exist in the ArmoniK glossary, this AEP seeks to foster community discussion and approval to ensure a unified understanding and usage of these terms.
+
+# Specification
+
+## Worker
+
+User-developed containerized software capable of performing one or several tasks depending on its implementation. A worker can simply take input data and perform calculations on it to return a result. A worker can also submit new tasks that will be self-performed or handled by different workers or other instances of itself.
+
+## Scheduling agent
+
+Containerized software cohabiting with a worker within a pod, running a specific algorithm to determine which tasks "its" worker (the one with which it shares the pod) should perform. It also manages all interactions between the worker and the databases (retrieving/saving data, creating new tasks, etc.), as well as managing worker errors and retrying/resubmitting failed tasks when necessary. A scheduling agent, like a worker, exists within a single partition.
+
+## Partition
+
+Logical segmentation of the Kubernetes cluster's pool of machines to distribute workloads according to usage. This feature is provided and handled by ArmoniK.
+
+## Task
+
+An atomic computation that takes one or several inputs and outputs one or several results. A task is launched by a client and processed by a worker. In ArmoniK, tasks cannot communicate with each other. They can, however, depend on each other via their input/output data, known as data dependency. 
```

Comment:
Tasks ability to submit other tasks is missing. Perhaps we could find a less abstract way of defining a task.